packages/tailwindcss/src/utils/topological-sort.ts

```diff
@@ -0,0 +1,36 @@
+export function topologicalSort<Key>(
+  graph: Map<Key, Set<Key>>,
+  options: { onCircularDependency: (path: Key[], start: Key) => void },
+): Key[] {
+  let seen = new Set<Key>()
+  let wip = new Set<Key>()
+
+  let sorted: Key[] = []
```

Comment:
We could make `topologicalSort` a Generator which has a nice API, but that's way slower compared to this array-based implementation:

<img width="1765" height="461" alt="Image" src="https://github.com/user-attachments/assets/3c419b82-0af5-42b8-b67b-fe78d0833e25" />

Reply:
Yeah I'd keep the array in this case. The perf tradeoff makes sense when the memory hit is possibly unbounded but you're likely to hit other perf and memory problems with large CSS files first.